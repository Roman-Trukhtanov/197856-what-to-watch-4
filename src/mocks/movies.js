const PREVIEW_URL = `http://placeimg.com/280/175/any`;
const BUCK_TRAILER_SRC = `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`;
const BUNNY_TRAILER_SRC = `https://www.radiantmediaplayer.com/media/big-buck-bunny-360p.mp4`;

export default [
  {
    id: 0,
    title: `Кремниевая долина`,
    preview: `${PREVIEW_URL}/1`,
    videoSrc: BUCK_TRAILER_SRC,
    videoType: `video/mp4`,
  },
  {
    id: 1,
    title: `Форсаж`,
    preview: `${PREVIEW_URL}/2`,
    videoSrc: BUNNY_TRAILER_SRC,
    videoType: `video/webm`,
  },
  {
    id: 2,
    title: `Жажда скорости`,
    preview: `${PREVIEW_URL}/3`,
    videoSrc: BUCK_TRAILER_SRC,
    videoType: `video/mp4`,
  },
  {
    id: 3,
    title: `Гарри Поттер`,
    preview: `${PREVIEW_URL}/4`,
    videoSrc: BUNNY_TRAILER_SRC,
    videoType: `video/webm`,
  },
  {
    id: 4,
    title: `Спецназ`,
    preview: `${PREVIEW_URL}/5`,
    videoSrc: BUCK_TRAILER_SRC,
    videoType: `video/mp4`,
  },
  {
    id: 5,
    title: `Игра престолов`,
    preview: `${PREVIEW_URL}/6`,
    videoSrc: BUNNY_TRAILER_SRC,
    videoType: `video/webm`,
  },
  {
    id: 6,
    title: `Стрела`,
    preview: `${PREVIEW_URL}/7`,
    videoSrc: BUCK_TRAILER_SRC,
    videoType: `video/mp4`,
  },
  {
    id: 7,
    title: `Флэш`,
    preview: `${PREVIEW_URL}/8`,
    videoSrc: BUNNY_TRAILER_SRC,
    videoType: `video/webm`,
  },
];
