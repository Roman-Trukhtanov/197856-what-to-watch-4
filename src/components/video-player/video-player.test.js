import React from "react";
import renderer from "react-test-renderer";
import VideoPlayer from "./video-player";
import movies from "../../mocks/movies";

const videoMock = movies[0];

describe(`VideoPlayer component`, () => {
  it(`Render VideoPlayer`, () => {
    const tree = renderer
      .create(<VideoPlayer
        isPlaying={true}
        videoType={videoMock.videoType}
        videoSrc={videoMock.videoSrc}
        preview={videoMock.preview}
      />, {
        createNodeMock(element) {
          if (element.type === `video`) {
            return {
              addEventListener() {},
            };
          }

          return null;
        }
      })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
