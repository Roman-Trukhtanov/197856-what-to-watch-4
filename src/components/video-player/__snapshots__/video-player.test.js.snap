// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer component Render VideoPlayer 1`] = `
<video
  controls={false}
  loop={true}
  muted={true}
  playsInline={true}
  poster="http://placeimg.com/280/175/any/1"
  preload="auto"
  style={
    Object {
      "height": "100%",
      "objectFit": "cover",
      "width": "100%",
    }
  }
>
  <source
    src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
    type="video/mp4"
  />
</video>
`;
